### PR TITLE
Gpl 387 run factory

### PR DESCRIPTION
**Pull request by Chris-Friend**
Changes proposed in this pull request:

* Add the run factory
* Ensure it only validates it's created entities once, and validation can be skipped altogether

### CODE REVIEW

File: app/models/ont/run_factory.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  # RunFactory
+  # The factory will create runs from a list of libraries
+  class RunFactory
+    include ActiveModel::Model
+
+    validate :check_run, :check_flowcells
+
+    def initialize(library_names = [])
+      @flowcells = []
+      build_run(library_names)
```
Comment (harrietc52) [inline app/models/ont/run_factory.rb:14]: I think the RunFactory is going to be initialized with libraryNames and position, a FlowcellType input, which includes the position. So `[{position: 1, library_name: 'aName'}, {position: 2, library_name: 'anotherName'}]`

File: app/models/ont/run_factory.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      @run.save(validate: false)
+      @flowcells.each { |flowcell| flowcell.save(validate: false) }
+      true
+    end
+
+    private
+
+    attr_reader :flowcells
+
+    def build_run(library_names)
```
Comment (harrietc52) [inline app/models/ont/run_factory.rb:31]: Which would change the was `idx` is used here as the position, as it is being passed in

File: config/pipelines.yml [hunk 3]
```diff
@@ -184,6 +184,7 @@ default: &default
     covid:
       external_study_id: "example id"
       species: "example species"
+      instrument_name: "GridIon"
```
Comment (harrietc52) [inline config/pipelines.yml:187]: "GridION"

File: app/models/ont/run_factory.rb (hunk 1 above)
Comment (stevieing) [reply app/models/ont/run_factory.rb:14]: Agreed. I thought that we would have a flowcell type with a position and a library name attribute.

**Pull request merged**